public/app/plugins/datasource/cloudwatch/components/LogGroupSelector.tsx

```diff
@@ -0,0 +1,163 @@
+import { MultiSelect } from '@grafana/ui';
+import { InputActionMeta } from '@grafana/ui/src/components/Select/types';
+import { notifyApp } from 'app/core/actions';
+import { createErrorNotification } from 'app/core/copy/appNotification';
+import { dispatch } from 'app/store/store';
+
+import { CloudWatchDatasource } from '../datasource';
+import { appendTemplateVariables } from '../utils/utils';
+
+const MAX_LOG_GROUPS = 20;
+const MAX_VISIBLE_LOG_GROUPS = 4;
+const DEBOUNCE_TIMER = 300;
+
+export interface LogGroupSelectorProps {
+  region: string;
+  selectedLogGroups: string[];
+  onChange: (logGroups: string[]) => void;
+
+  datasource?: CloudWatchDatasource;
+  allowCustomValue?: boolean;
+  onRunQuery?: () => void;
+  onOpenMenu?: () => Promise<void>;
+  refId?: string;
+  width?: number | 'auto';
+  saved?: boolean;
+}
+
+export const LogGroupSelector: React.FC<LogGroupSelectorProps> = ({
+  region,
+  selectedLogGroups,
+  onChange,
+  datasource,
+  allowCustomValue,
+  onRunQuery,
+  onOpenMenu,
+  refId,
+  width,
+  saved = true,
+}) => {
+  const [loadingLogGroups, setLoadingLogGroups] = useState(false);
+  const [availableLogGroups, setAvailableLogGroups] = useState<Array<SelectableValue<string>>>([]);
+  const logGroupOptions = useMemo(
+    () => unionBy(availableLogGroups, selectedLogGroups?.map(toOption), 'value'),
+    [availableLogGroups, selectedLogGroups]
+  );
+
+  const fetchLogGroupOptions = useCallback(
+    async (region: string, logGroupNamePrefix?: string) => {
+      if (!datasource) {
+        return [];
+      }
+      try {
+        const logGroups: string[] = await datasource.describeLogGroups({
+          refId,
+          region,
+          logGroupNamePrefix,
+        });
+        return logGroups.map(toOption);
+      } catch (err) {
+        let errMessage = 'unknown error';
+        if (typeof err !== 'string') {
+          try {
+            errMessage = JSON.stringify(err);
+          } catch (e) {}
+        } else {
+          errMessage = err;
+        }
+        dispatch(notifyApp(createErrorNotification(errMessage)));
+        return [];
+      }
+    },
+    [datasource, refId]
+  );
+
+  const onLogGroupSearch = async (searchTerm: string, region: string, actionMeta: InputActionMeta) => {
+    if (actionMeta.action !== 'input-change') {
+      return Promise.resolve();
+    }
+    if (!datasource) {
+      return Promise.resolve();
+    }
+
+    // No need to fetch matching log groups if the search term isn't valid
+    // This is also useful for preventing searches when a user is typing out a log group with template vars
+    // See https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_LogGroup.html for the source of the pattern below
+    const logGroupNamePattern = /^[\.\-_/#A-Za-z0-9]+$/;
+    if (!logGroupNamePattern.test(searchTerm)) {
+      return Promise.resolve();
+    }
+
+    setLoadingLogGroups(true);
+    try {
```

Comment:
fetchLogGroupOptions already catches error, I don't think you need another try here.